File: back/src/main/java/br/com/bagnascojhoel/portfolio_website_bff/model/project/Project.java
package br.com.bagnascojhoel.portfolio_website_bff.model.project;

import br.com.bagnascojhoel.portfolio_website_bff.model.extra_portfolio_description.Complexity;
import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.annotation.Nullable;
import lombok.*;

import java.time.Instant;
import java.util.Set;

@Builder
@AllArgsConstructor
@Getter
@Data
public class Project {
    @NonNull
    @JsonProperty("uniqueName")
    private String repositoryId;

    @NonNull
    private String title;

    @NonNull
    private String description;

    @NonNull
    private Set<String> tags;

    @NonNull
    private String repositoryUrl;

    @Nullable
    private String websiteUrl;

    @NonNull
    private Boolean startsOpen;

    @NonNull
    private Complexity complexity;

    @NonNull
    private Instant lastChangedDateTime;

    @JsonGetter("repositoryId")
    public String getRepositoryId() {
        return repositoryId;
    }
}

File: front/src/domain/Project.ts
export type Project = {
    projectId: string;
    title: string;
    description: string;
    tags: string[];
    githubUrl: string;
    websiteUrl?: string;
    complexity: Complexity;
    lastChangedDateTime: Date;
    startsOpen: boolean;
};

type Complexity = {
    code: string;
    text: string;
};

export const ComplexityCode = {
    'complexity-extreme': 3,
    'complexity-high': 2,
    'complexity-medium': 1,
    'complexity-low': 0,
};
